src/tests/application/verifier_tests.cpp

```diff
@@ -0,0 +1,43 @@
+#include <gtest/gtest.h>
+#include "AbstractDatabaseInterfaceMock.h"
+#include "../../application/EntryRepository.h"
+#include "../../application/PasswordVerifier.h"
+#include "../../application/InstanceManager.h"
+#include "../../plugins/encryption/DefaultHash.h"
+#include "../../application/Hash.h"
+#include <stdexcept>
+
+using namespace DDD;
+
+ACTION(ThrowDomainError) {
+    throw std::domain_error("");
+}
+
+TEST(SetMasterPassword, DDDTests) {
+    AbstractDatabaseInterfaceMock mock;
+
+    Hash* hash = new DefaultHash();
+
+    std::string masterPassword("masterPassword");
+    ValueObjects::PlaintextPassword plainMasterPassword(masterPassword);
+
+    ValueObjects::EncryptedPassword hashedPassword(hash->compute(hash->compute(masterPassword)));
+
+    Services::MasterPasswordVerifier verifier(hash);
+
+    const DatabaseColumn c = DatabaseColumn::EncryptedPassword;
+
+    ValueObjects::EntryId id(0);
+
+    Entities::Entry e(id, ValueObjects::EntryName("masterPasswordHash"), ValueObjects::Login("master"), hashedPassword);
+
+    Repositories::EntryRepository* repo = new Repositories::EntryRepository(&mock);
```

Comment:
Why is this allocated on the heap?

Reply:
Might not be necessary, but does it really matter in a test function? I should hope not.

Reply:
I was just genuinely curious as my go-to is always the stack at first.